pyomo/contrib/parmest/utils/model_utils.py

```diff
@@ -69,20 +73,24 @@ def convert_params_to_vars(model, param_names=None, fix_vars=False):
             substitution_map[id(theta_object)] = theta_var_object
             comp_map[theta_object] = theta_var_object
 
-        # Indexed Param
+        # Indexed Param -- Delete Param, add Var
         elif isinstance(theta_object, IndexedParam):
-            # Delete Param, add Var
-            # Before deleting the Param, create a list of the indexed param names
+
+            # save Param values
             vals = theta_object.extract_values()
-            param_theta_objects = []
-            for theta_obj in theta_object:
-                indexed_param_name = theta_object.name + '[' + str(theta_obj) + ']'
-                theta_cuid = ComponentUID(indexed_param_name)
-                param_theta_objects.append(theta_cuid.find_component_on(model))
-                indexed_param_names.append(indexed_param_name)
 
+            # get indexed Params
+            param_theta_objects = [theta_obj for _, theta_obj in theta_object.items()]
+
+            # get indexed Param names
+            indexed_param_CUIDs += [
+                ComponentUID(theta_obj) for _, theta_obj in theta_object.items()
+            ]
```

Comment:
It would be more clear / more efficient to use list's `extend()` and dict's `values()`:
```python
indexed_param_CUIDs.extend(
    ComponentUID(theta_obj) for theta_obj in theta_object.values()
)
```

Reply:
Changed as suggested.